File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/database/MediaRealmDataStore.java
package com.lattis.ellipse.data.database;

import com.lattis.ellipse.data.database.base.RealmObservable;
import com.lattis.ellipse.data.database.mapper.RealmMediaMapper;
import com.lattis.ellipse.data.database.model.RealmMedia;
import com.lattis.ellipse.domain.model.Media;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.realm.RealmConfiguration;

/**
 * Created by raverat on 4/6/17.
 */

public class MediaRealmDataStore {

    private RealmMediaMapper mediaMapper;
    private RealmConfiguration realmConfiguration;

    @Inject
    public MediaRealmDataStore(RealmConfiguration realmConfiguration, RealmMediaMapper mediaMapper) {
        this.mediaMapper = mediaMapper;
        this.realmConfiguration=realmConfiguration;
    }

    public Observable<Media> getMedia(String id) {
        return RealmObservable.object(
                realm -> {
                    RealmMedia realmMedia = realm.where(RealmMedia.class).equalTo("id", id).findFirst();
                    return realmMedia != null ? realmMedia : new RealmMedia();
                })
                .map(realmMedia -> mediaMapper.mapOut(realmMedia));
    }

    public Observable<Media> createOrUpdateUser(Media media) {
        return RealmObservable.object(realm -> realm.copyToRealmOrUpdate(mediaMapper.mapIn(media)))
                .map(realmMedia -> mediaMapper.mapOut(realmMedia));
    }

}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/database/store/MediaRealmDataStore.kt
package com.lattis.data.database.store


import com.lattis.data.database.base.RealmObservable
import com.lattis.data.database.mapper.RealmMediaMapper
import com.lattis.data.database.model.RealmMedia
import com.lattis.domain.models.Media
import javax.inject.Inject

import io.reactivex.rxjava3.core.Observable
import io.realm.RealmConfiguration
import io.reactivex.rxjava3.functions.Function

/**
 * Created by raverat on 4/6/17.
 */

class MediaRealmDataStore @Inject
constructor(private val mediaMapper: RealmMediaMapper) {

    fun getMedia(id: String): Observable<Media> {
        return RealmObservable.`object`<RealmMedia>(
                Function { realm ->
                    val realmMedia = realm.where(RealmMedia::class.java).equalTo("id", id).findFirst()
                    if (realmMedia != null) realmMedia else RealmMedia()
                })
                .map { realmMedia -> mediaMapper.mapOut(realmMedia) }
    }

    fun createOrUpdateUser(media: Media): Observable<Media> {
        return RealmObservable.`object`<RealmMedia>(
                Function{ realm -> realm.copyToRealmOrUpdate(mediaMapper.mapIn(media)) })
                .map { realmMedia -> mediaMapper.mapOut(realmMedia) }
    }

}
